### PR TITLE
feat(lint): Add Sass Linting to the test:lint pipeline

### LINKED ISSUE #75: Add some Sass linting?
I have a local, crude lint set up using https://github.com/sasstools/sass-lint and have it returning about 74 warnings and 4 errors (mostly sorting violations and nit-picky formatting)...

### .sass-lint.yml
```yml
options:
  merge-default-rules: true

files:
  include: 'content-src/**/*.scss'


rules:
  extends-before-declarations: 2
  extends-before-mixins: 2
  hex-notation: [2, {style: uppercase}]
  indentation: [2, {size: 2}]
  leading-zero: [2, {include: true}]
  mixins-before-declarations: [2, {exclude: [breakpoint, mq]}]
  nesting-depth: [2, {max-depth: 4}]
  no-debug: 1
  no-ids: 1
  no-important: 1
  no-warn: 1
  placeholder-in-extend: 2
  property-sort-order: [1, {order: [display, margin], ignore-custom-properties: true}]
```

After you do an `npm i sass-lint` locally, you can run the following command to see the lint results:
```sh
$ ./node_modules/.bin/sass-lint -v -q #or just `sass-lint -v -q` in the package.json scripts section
```

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #75

### CODE REVIEW

**Comment by pdehaan** [comment]
Feel free to smite this down.
I got it down to ~22 warnings and 0 errors, and `npm start` seemed to look the same to my untrained eyes.

Current output is:
```
> activity-streams@0.0.1 test:lint /Users/pdehaan/dev/github/activity-streams
> eslint . && jscs . && sass-lint -v -q


content-src/components/Header/Header.scss
   3:15  warning  Color literals such as 'FFF' should only be used in variable declarations         no-color-literals
   4:28  warning  Color literals such as 'DFE0E1' should only be used in variable declarations      no-color-literals
   7:3   warning  Duplicate properties are not allowed within a block                               no-duplicate-properties
  23:19  warning  Color literals such as 'F3F4F6' should only be used in variable declarations      no-color-literals
  24:38  warning  Color literals such as 'F3F4F6' should not be passed to functions, use variables  no-color-literals
  52:15  warning  Color literals such as 'F3F4F6' should only be used in variable declarations      no-color-literals
  53:28  warning  Color literals such as 'F3F4F6' should not be passed to functions, use variables  no-color-literals
  71:26  warning  Color literals such as 'F3F4F6' should not be passed to functions, use variables  no-color-literals

content-src/components/NewTabPage/NewTabPage.scss
  1:5  warning  Qualifying elements are not allowed for class selectors       no-qualifying-elements
  1:5  warning  Class `.new-tab` should be nested within its parent selector  force-element-nesting

content-src/components/TimelinePage/TimelinePage.scss
   1:5   warning  Qualifying elements are not allowed for class selectors                       no-qualifying-elements
   1:5   warning  Class `.timeline` should be nested within its parent selector                 force-element-nesting
   9:6   warning  Qualifying elements are not allowed for class selectors                       no-qualifying-elements
   9:6   warning  Class `.sidebar` should be nested within its parent selector                  force-element-nesting
  11:12  warning  Color literals such as '888B92' should only be used in variable declarations  no-color-literals
  34:32  warning  Color literals such as 'E14164' should only be used in variable declarations  no-color-literals
  39:10  warning  Qualifying elements are not allowed for class selectors                       no-qualifying-elements
  39:10  warning  Class `.content` should be nested within its parent selector                  force-element-nesting

content-src/components/TopSites/TopSites.scss
  29:25  warning  Color literals such as 'FFF' should only be used in variable declarations  no-color-literals
  48:13  warning  Class `.tile-title` should be nested within its parent pseudoClass         force-element-nesting

content-src/styles/normalize.scss
  6:2  warning  Pseudo-class `:before` should be nested within its parent selector  force-pseudo-nesting
  7:2  warning  Pseudo-class `:after` should be nested within its parent selector   force-pseudo-nesting

✖ 22 problems (0 errors, 22 warnings)
```

We can tweak the rules until it works for everybody. Most of them are fairly straight foward (use color variables instead of hard-coding values throughout), but there was one interesting result which may be a bug (or not):

> #### Duplicate properties are not allowed within a block 
> &mdash; via [./content-src/components/Header/Header.scss:2-4](https://github.com/mozilla/activity-streams/blob/9a01549185b920ce9690af4a3173584f376f6dae/content-src/components/Header/Header.scss#L2-L4)

**Comment by k88hudson** [comment]
This is awesome! Tests are failing right now due to an unrelated issue with the add-on tests, but this would be :+1: to add

**Comment by k88hudson** [comment]
Let's try to merge this next week

**Comment by pdehaan** [comment]
Current, current output:

```
> eslint . && jscs . && sass-lint -v -q

content-src/components/ActivityFeed/ActivityFeed.scss
  55:16  warning  Pseudo-element `::after` should be nested within its parent pseudoClass  force-pseudo-nesting
  71:11  warning  Class `.feed-icon` should be nested within its parent class              force-element-nesting

content-src/components/Header/Header.scss
   3:15  warning  Color literals such as 'FFF' should only be used in variable declarations         no-color-literals
   4:28  warning  Color literals such as 'DFE0E1' should only be used in variable declarations      no-color-literals
   7:3   warning  Duplicate properties are not allowed within a block                               no-duplicate-properties
  37:19  warning  Color literals such as 'F3F4F6' should only be used in variable declarations      no-color-literals
  38:38  warning  Color literals such as 'F3F4F6' should not be passed to functions, use variables  no-color-literals
  73:15  warning  Color literals such as 'F3F4F6' should only be used in variable declarations      no-color-literals
  74:28  warning  Color literals such as 'F3F4F6' should not be passed to functions, use variables  no-color-literals
  92:26  warning  Color literals such as 'F3F4F6' should not be passed to functions, use variables  no-color-literals

content-src/components/NewTabPage/NewTabPage.scss
  1:5  warning  Qualifying elements are not allowed for class selectors       no-qualifying-elements
  1:5  warning  Class `.new-tab` should be nested within its parent selector  force-element-nesting

content-src/components/SiteIcon/SiteIcon.scss
  15:12  warning  Color literals such as 'FFF' should only be used in variable declarations  no-color-literals

content-src/components/Spotlight/Spotlight.scss
  14:15  warning  Color literals such as 'FFF' should only be used in variable declarations               no-color-literals
  37:13  warning  Property `margin` should be written more concisely as `0 0 5px` instead of `0 0 5px 0`  shorthand-values

content-src/components/TimelinePage/TimelinePage.scss
   1:5   warning  Qualifying elements are not allowed for class selectors                       no-qualifying-elements
   1:5   warning  Class `.timeline` should be nested within its parent selector                 force-element-nesting
   9:6   warning  Qualifying elements are not allowed for class selectors                       no-qualifying-elements
   9:6   warning  Class `.sidebar` should be nested within its parent selector                  force-element-nesting
  11:12  warning  Color literals such as '888B92' should only be used in variable declarations  no-color-literals
  30:16  warning  Color literals such as 'FFF' should only be used in variable declarations     no-color-literals
  40:23  warning  Color literals such as 'E14164' should only be used in variable declarations  no-color-literals
  45:10  warning  Qualifying elements are not allowed for class selectors                       no-qualifying-elements
  45:10  warning  Class `.content` should be nested within its parent selector                  force-element-nesting

content-src/components/TopSites/TopSites.scss
  35:13  warning  Class `.tile-title` should be nested within its parent pseudoClass  force-element-nesting

content-src/main.scss
  32:4  warning  Class `.fa-timeline` should be nested within its parent class  force-element-nesting

content-src/styles/normalize.scss
  6:2  warning  Pseudo-class `:before` should be nested within its parent selector  force-pseudo-nesting
  7:2  warning  Pseudo-class `:after` should be nested within its parent selector   force-pseudo-nesting

✖ 28 problems (0 errors, 28 warnings)
```

Mostly just reminders that we should be pulling color values outside of the individual Sass files and into the variables file, and then a couple on nesting (we could probably disable those checks since I think adding another level of nesting would make the code a bit uglier), and then one interesting bit on duplicate CSS property declaration which is suspicious.

File: content-src/components/Header/Header.scss [hunk 1]
```diff
@@ -1,28 +1,31 @@
 .head {
-  display: relative;
-  z-index: 1;
+  align-items: center;
+  background: #FFF;
+  border-bottom: solid 1px #DFE0E1;
+  box-shadow: $header-box-shadow;
   display: flex;
+  display: relative;
```
Comment (pdehaan) [inline content-src/components/Header/Header.scss]: > [7:3]   **warning**  Duplicate properties are not allowed within a block

**Comment by k88hudson** [comment]
:+1: :fireworks: 

High five

**Pull request merged**